src/main/java/com/google/sps/utility/GmailResponseUtility.java

```diff
@@ -0,0 +1,252 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.utility;
+
+import com.google.api.services.gmail.model.Message;
+import com.google.api.services.gmail.model.MessagePartHeader;
+import com.google.sps.exceptions.GmailMessageFormatException;
+import com.google.sps.model.GmailClient;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/** Contains business logic to calculate statistics for the GmailResponse */
+public final class GmailResponseUtility {
+  private GmailResponseUtility() {}
+
+  /**
+   * Get list of unread emails from last nDays days from user's Gmail account
+   *
+   * @param gmailClient GmailClient implementation with valid google credential
+   * @param messageFormat GmailClient.MessageFormat setting to control how much of each message is
+   *     returned
+   * @return List of unread messages from last nDays from user's Gmail account with requested level
+   *     of information
+   * @throws IOException if an issue occurs with the Gmail service
+   */
+  public static List<Message> getUnreadEmailsFromNDays(
+      GmailClient gmailClient, GmailClient.MessageFormat messageFormat, int nDays)
+      throws IOException {
+    String searchQuery = GmailClient.emailQueryString(nDays, "d", true, false, "");
+    return gmailClient.listUserMessages(searchQuery).stream()
+        .map(
+            (message) -> {
+              try {
+                return gmailClient.getUserMessage(message.getId(), messageFormat);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            })
+        .collect(Collectors.toList());
+  }
```

Comment:
I think this method belongs on `GmailClient` instead. No other method in this utility depends on the client - this one stands out.

Reply:
Agreed - done!